### PR TITLE
MQE: Use kahan in sum aggregation

### PR DESCRIPTION
**Pull request by jhesketh**
Implements the improvements from
https://github.com/prometheus/prometheus/pull/14074 https://github.com/prometheus/prometheus/pull/14362

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/streamingpromql/operators/aggregation.go [hunk 1]
```diff
@@ -96,6 +97,7 @@ type group struct {
 
 	// Sum, presence, and histograms for each step.
 	floatSums           []float64
+	floatMeans          []float64 // Mean, or "compensating value" for Kahan summation.
```
Comment (charleskorn) [inline pkg/streamingpromql/operators/aggregation.go:100]: Would `floatCompensatingValues` or something like that be a clearer name?

File: pkg/streamingpromql/testdata/upstream/aggregators.test [hunk 2]
```diff
@@ -602,8 +602,8 @@ load 10s
 	data{test="nan",group="2",point="b"} NaN
 
 # Unsupported by streaming engine.
```
Comment (charleskorn) [inline pkg/streamingpromql/testdata/upstream/aggregators.test:604]: We can remove this comment now 🙂

Comment (seizethedave) [inline pkg/streamingpromql/testdata/upstream/aggregators.test:604]: is it now supported?

File: pkg/streamingpromql/operators/aggregation.go (hunk 1 above)
Comment (jhesketh) [reply pkg/streamingpromql/operators/aggregation.go:100]: Perhaps. We may use this for mean's in the future, but will rename for now (can always go back)

**Review by charleskorn — APPROVED** [review]

**Pull request merged**